src/components/Form/Combobox.js

```diff
@@ -0,0 +1,333 @@
+
+        if (count > 1) {
+          return `${count} ${labelMultiselect}`;
+        } else if (count === 1) {
+          return selectedItems[0].label;
+        }
+      } else {
+        return value;
+      }
+
+      return '';
+    };
+
+    const label = formatLabel();
+
+    const classNames = cx(
+      'slds-combobox__input',
+      { 'slds-combobox__input-value': !!label },
+    );
+
+    const iconRight = readOnly ? 'down' : 'search';
+
+    return (
+      <InputRaw
+        aria-controls={`listbox-${id}`}
+        autoComplete="off"
+        className={classNames}
+        disabled={disabled}
+        iconRight={iconRight}
+        iconRightOnClick={this.onToggle}
+        id={`combobox-${id}`}
+        onChange={onChange}
+        onClick={this.onToggle}
+        onKeyDown={readOnly ? null : this.onInputKeyDown}
+        placeholder={placeholder}
+        readOnly={readOnly}
+        required={required}
+        role="textbox"
+        value={label}
+      />
+    );
+  }
+
+  renderComboboxItems() {
+    const { items } = this.props;
+
+    return items.map(item => (
+      <ComboboxDropdownListItem
+        icon={item.icon}
+        id={`listbox-option-${item.key}`}
+        isHeader={item.isHeader}
+        key={item.key}
+        onClick={this.getSelectHandler(item.key)}
+        selected={item.selected}
+      >
+        {item.label}
+      </ComboboxDropdownListItem>
+    ));
+  }
+
+  renderInlineListbox() {
+    const selectedItems = this.getSelectedItems();
+
+    return (
+      <Listbox className="slds-listbox_inline">
+        {selectedItems.map(({ icon, key, label }) => (
+          <Pill
+            icon={icon ? <Icon icon={icon.icon} size="small" sprite={icon.sprite} /> : null}
+            key={key}
+            label={label}
+            title={label}
+            onClose={this.getSelectHandler(key)}
+          />
+        ))}
+      </Listbox>
+    );
+  }
+
+  renderSelectedPills() {
+    const { onSelect } = this.props;
+
+    const selectedItems = this.getSelectedItems();
+
+    return (
+      <Listbox className="slds-listbox_inline slds-p-top_xxx-small">
+        {selectedItems.map(({ icon, key, label }) => (
+          <Pill
+            icon={icon ? <Icon icon={icon.icon} size="small" sprite={icon.sprite} /> : null}
+            key={key}
+            label={label}
+            title={label}
+            onClose={() => onSelect(key)}
+          />
+        ))}
+      </Listbox>
+    );
+  }
+
+  render() {
+    const {
+      className,
+      closeOnClickOutside,
+      error,
+      height,
+      hideLabel,
+      id,
+      inlineListbox,
+      labelInput,
+      readOnly,
+      required,
+      size,
+    } = this.props;
+
+    const { open } = this.state;
+
+    const condition = closeOnClickOutside && open;
+
+    const renderPills = !inlineListbox && (
+      (readOnly && this.getSelectedItems().length > 1) ||
+      (!readOnly && this.getSelectedItems().length > 0)
```

Comment:
This is the third time `getSelectedItems` is computed, it might make sense to store in a const